File: Logic_HandsOn/Java/Problems on Array,Pattern etc/Program30.java

///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		a program which accept accept range from user and display all numbers in between that range in reverse order.
//
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.util.*;

class Solution
{
	public int iStart,iEnd;

	public Solution(int iStart,int iEnd)
	{
		this.iStart = iStart;
		this.iEnd = iEnd;
	}

	public void RangeDisplayRev()
	{
                        
		for(int iCnt=iEnd;iCnt>=iStart;iCnt--)
		{
			System.out.print(iCnt+" ");
		}

		System.out.println();
	}
}

class Program
{
	public static void main(String[] args)
	{
		Scanner sObj = new Scanner(System.in);

		System.out.println("Enter Start: ");
		int iStart = sObj.nextInt();

		System.out.println("Enter End: ");
		int iEnd = sObj.nextInt();
		
		Solution obj = new Solution(iStart,iEnd);
		obj.RangeDisplayRev();
	}
}

/*
OUTPUT:

Enter Start: 
23
Enter End: 
35
35 34 33 32 31 30 29 28 27 26 25 24 23 

*/

File: Logic_HandsOn/C++/Problems on Digits,String,Array,Pointers etc/program30.cpp
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		a program which accept accept range from user and display all numbers in between that range in reverse order.
//
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

#include<iostream>
using namespace std;

class Range
{
	private:
		int iStart;
		int iEnd;
		int iCnt;

	public:
		Range(int iStart=0,int iEnd=0,int iCnt=0)
		{
			this->iStart = iStart;
			this->iEnd = iEnd;
		}	

		void RangeDisplayRev()
		{
			if (iStart>iEnd)
			{
				cout<<"Invalid Input";	
			}
			else
			{       
		       		for(iCnt=iEnd;iCnt>=iStart;iCnt--)
 		               	{  
                 		       printf("%d ",iCnt);
                		}

			}	
			cout<<"\n";
		}
};


int main()
{
	int iValue1=0,iValue2=0;

	cout<<"Enter starting point ";
	cin>>iValue1;
	
	cout<<"Enter ending point ";
	cin>>iValue2;
	
	Range *rObj = new Range(iValue1,iValue2);
	rObj->RangeDisplayRev();

	return 0;
}

